woocommerce/woocommerce-ios
# Blaze: Add picker view for target languages

Closes: #11630 


## Description

This PR adds a new screen for selecting target languages for a creating a new Blaze campaign. Changes include:
- Updated all Blaze target option models to conform to `Hashable` and `Identifiable`. This is necessary to display the models on a SwiftUI list.
- Renamed `SelectionList` to `SingleSelectionList`. For simplicity, I created a separate reusable view `MultiSelectionList` for use in the Blaze campaign creation flow.
- Created `BlazeTargetLanguagePickerView` and view model to sync and display languages for picking.
- Integrated the language picker view to the Blaze campaign creation form.

## Testing instructions

- Log in to a store with at least one existing product and is eligible for Blaze.
- Navigate to the Products tab and select any public product.
- On the product form, select Promote with Blaze.
- Notice that the campaign creation form is presented. Select the language field.
- Notice that a new screen is displayed for selecting target languages. Confirm that the selection and filtering work as expected.
- Select Cancel, the view should be dismissed without any changes to the campaign creation form.
- Select the language field again, this time select a different option and tap Save. The language field in the creation form should be updated correctly.


## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/5533851/a4a12605-5e74-4c03-a0a5-fc3d4876d608



---
